pkg/install/genevalogging.go

```diff
@@ -0,0 +1,399 @@
+	  CONTAINER_ID ${record["CONTAINER_NAME"].nil? ? nil : record["CONTAINER_NAME"].split("_")[4] }
+	</record>
+  </filter>
+  <match **>
+	@type mdsd
+	acktimeoutms 0
+	buffer_type memory
+	buffer_queue_full_action block
+	disable_retry_limit true
+	djsonsocket /var/run/mdsd/default_djson.socket
+	emit_timestamp_name time
+	flush_interval 10s
+  </match>
+`
+)
+
+func mdsdConfig(location, resourceGroup, subscriptionID, clusterName, genevaLoggingNamespace, genevaLoggingAccount string) (string, error) {
+	var tmpl = `<?xml version="1.0" encoding="utf-8"?>
+	    <MonitoringManagement version="1.0" namespace="{{ .Namespace | Escape }}" eventVersion="1" timestamp="2017-08-01T00:00:00.000Z">
+	        <Accounts>
+	            <Account moniker="{{ .Account | Escape }}" isDefault="true" autoKey="false"/>
+	        </Accounts>
+	        <Management eventVolume="Large" defaultRetentionInDays="90">
+	            <Identity tenantNameAlias="ResourceName" roleNameAlias="ResourceGroupName" roleInstanceNameAlias="SubscriptionId">
+	                <IdentityComponent name="Region">{{ .Region | Escape }}</IdentityComponent>
+	                <IdentityComponent name="SubscriptionId">{{ .SubscriptionId | Escape }}</IdentityComponent>
+	                <IdentityComponent name="ResourceGroupName">{{ .ResourceGroupName | Escape }}</IdentityComponent>
+	                <IdentityComponent name="ResourceName">{{ .ResourceName | Escape }}</IdentityComponent>
+	            </Identity>
+	            <AgentResourceUsage diskQuotaInMB="50000"/>
+	        </Management>
+	        <Sources>
+	            <Source name="journald" dynamic_schema="true"/>
+	            <Source name="audit" dynamic_schema="true"/>
+	        </Sources>
+	        <Events>
+	            <MdsdEvents>
+	                <MdsdEventSource source="journald">
+	                    <RouteEvent eventName="CustomerSyslogEvents" storeType="CentralBond" priority="High"/>
+	                </MdsdEventSource>
+	                <MdsdEventSource source="audit">
+	                    <RouteEvent eventName="CustomerAuditLogEvents" storeType="CentralBond" priority="High"/>
+	                </MdsdEventSource>
+	            </MdsdEvents>
+	        </Events>
+		</MonitoringManagement>`
+
+	t := template.Must(template.New("").Funcs(map[string]interface{}{
+		"Escape": func(s string) (string, error) {
+			var b bytes.Buffer
+			err := xml.EscapeText(&b, []byte(s))
+			return b.String(), err
+		},
+	}).Parse(tmpl))
+
+	b := &bytes.Buffer{}
+
+	err := t.Execute(b, map[string]string{
+		"Namespace":         genevaLoggingNamespace,
+		"Account":           genevaLoggingAccount,
+		"Region":            location,
+		"SubscriptionId":    subscriptionID,
+		"ResourceName":      clusterName,
+		"ResourceGroupName": resourceGroup,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	return string(b.Bytes()), nil
+}
+
+func (i *Installer) installGenevaLogging(ctx context.Context) error {
```

Comment:
function needs to be reentrant, think "ensureGenevaLogging".  Probably worth starting all of this *outside* pkg/install so that we can call it on adminupdate as well if we want to and pull it out to an operator when we're ready